src/RealSetupManager.cpp

```diff
@@ -51,6 +53,25 @@ std::vector< std::pair<double, double> > RealSetupManager::getHVPMTValue() {
     return hv_values;
 }
 
+bool RealSetupManager::propagateDiscriSettings() {
+    // Take action together with checking if the action succeded 
+    // NB : Setup functions return several different int's so far
+    
+    bool succeeded_majority = (m_discri.setMajority(m_interface.getConditions().getChannelsMajority()) == 1);
+
+    bool succeeded_discriSettings = true;
+    for (int dc_id = 0; dc_id < m_interface.getConditions().getNDiscriChannels(); dc_id++) {
+        succeeded_discriSettings = (succeeded_discriSettings && ((m_discri.setChannel(dc_id, m_interface.getConditions().getDiscriChannelState(dc_id))) == 1));
+        succeeded_discriSettings = (succeeded_discriSettings && (m_discri.setTh(m_interface.getConditions().getDiscriChannelThreshold(dc_id), dc_id) == 1));
+        succeeded_discriSettings = (succeeded_discriSettings && ((m_discri.setWidth(m_interface.getConditions().getDiscriChannelWidth(dc_id), dc_id)) == 1));
+        succeeded_discriSettings = (succeeded_discriSettings && ((m_discri.setWidth(m_interface.getConditions().getDiscriChannelWidth(dc_id), dc_id)) == 1));
```

Comment:
All the parentheses here are not needed (operator == has precedence over &&), could be removed...